### PR TITLE
Shrink anthology.bib (closes #3016)

### LINKED ISSUE #3016: Compact bibfile
Overleaf has a 50 MB file size limit, and `anthology.bib` is now larger than this. We should create a compact BibTeX export using string substitution [as suggested here](https://twitter.com/daniel_hers/status/1744434842895294496). I'm not sure if this should just replace the current Anthology bib file, or become a new export, say `anthology-compact.bib`:

* The advantage of encompactifying the current file is it would work for everyone without having to change anything.
* The disadvantage is it complicates cutting-and-pasting.
* However, we already have [https://aclanthology.org/anthology+abstracts.bib.gz](https://aclanthology.org/anthology+abstracts.bib.gz). I'm not sure how often people cut-and-paste individual entries from the complete file, anyway; it seems that it's main use is in Overleaf.

I'm therefore include to simply replace `anthology.bib`.

### PR DESCRIPTION
**Pull request by mjpost**
This gets `anthology.bib` to 41 MB. To make it smaller, I

* Removed newlines and extra spaces from each entry
* Used two global string substitutions, and another per-volume one for the book title

### CODE REVIEW

**Comment by mjpost** [comment]
An example of the generated bib file [can be found here](https://cs.jhu.edu/~post/tmp/anthology-shrunk.bib). Here are a few example entries:

```bibtex
@string{acl = {Association for Computational Linguistics}}
@string{anth = {https://aclanthology.org/}}

@string{YRRSDS:2023:1 = {Proceedings of the 19th Annual Meeting of the Young Reseachers' Roundtable on Spoken Dialogue Systems}}
@proceedings{yrrsds-2023-young, title = "Proceedings of the 19th Annual Meeting of the Young Reseachers' Roundtable on Spoken Dialogue Systems", editor = "Hudecek, Vojtech and Schmidtova, Patricia and Dinkar, Tanvi and Chiyah-Garcia, Javier and Sieinska, Weronika", month = sep, year = "2023", address = "Prague, Czechia", publisher = acl, url = anth # {2023.yrrsds-1.0}, }
@inproceedings{chiyah-garcia-2023-processing, title = "Processing Referential Ambiguities in Situated Dialogue Systems", author = "Chiyah-Garcia, Javier", editor = "Hudecek, Vojtech and Schmidtova, Patricia and Dinkar, Tanvi and Chiyah-Garcia, Javier and Sieinska, Weronika", booktitle = YRRSDS:2023:1, month = sep, year = "2023", address = "Prague, Czechia", publisher = acl, url = anth # {2023.yrrsds-1.1}, pages = "1--4", }
@inproceedings{dinkar-2023-safety, title = "Safety and Robustness in Conversational {AI}", author = "Dinkar, Tanvi", editor = "Hudecek, Vojtech and Schmidtova, Patricia and Dinkar, Tanvi and Chiyah-Garcia, Javier and Sieinska, Weronika", booktitle = YRRSDS:2023:1, month = sep, year = "2023", address = "Prague, Czechia", publisher = acl, url = anth # {2023.yrrsds-1.2}, pages = "5--8", }
@inproceedings{addlesee-2023-incremental, title = "Incremental Speech Processing for Voice Assistant Accessibility", author = "Addlesee, Angus", editor = "Hudecek, Vojtech and Schmidtova, Patricia and Dinkar, Tanvi and Chiyah-Garcia, Javier and Sieinska, Weronika", booktitle = YRRSDS:2023:1, month = sep, year = "2023", address = "Prague, Czechia", publisher = acl, url = anth # {2023.yrrsds-1.3}, pages = "9--11", }
@inproceedings{colabianchi-2023-advancing, title = "Advancing Spoken Dialog Systems for Manufacturing: From Conceptual Architecture and Taxonomy to Real Case Applications and Future Directions", author = "Colabianchi, Silvia", editor = "Hudecek, Vojtech and Schmidtova, Patricia and Dinkar, Tanvi and Chiyah-Garcia, Javier and Sieinska, Weronika", booktitle = YRRSDS:2023:1, month = sep, year = "2023", address = "Prague, Czechia", publisher = acl, url = anth # {2023.yrrsds-1.4}, pages = "12--14", }
```

**Review by mbollmann — APPROVED** [review]
LGTM! I was wondering how I feel about doing this manipulation on the generated BibTeX strings, rather than at the source where we generate them, but maybe we should be pragmatic here and do the thing that is quicker to implement. The implementation is also not that long nor hard to follow.

I also played around very briefly with the compacted bib file in Overleaf with different TeX engines and it didn't produce any immediate hiccups either.

**Comment by mjpost** [comment]
Yeah, it's definitely a hack. I tried a deeper, correcter approach, but it was a bit more complicated and I'm also wary of adding to the old code base. We should prioritize switching to the new one first, and this can serve as a quick-fix in the meantime. Thanks for the sanity check, too.

**Comment by joanise** [comment]
This is a great idea! I have a question, though, about maintaining human-readability: the fact that it's one record per line in this format means it's more difficult to read. Could we consider using `,\n\t` instead of `, ` as the field separator, making the file only a little bit bigger but keeping it easier to read?

**Comment by mbollmann** [comment]
> This is a great idea! I have a question, though, about maintaining human-readability: the fact that it's one record per line in this format means it's more difficult to read. Could we consider using `,\n\t` instead of `, ` as the field separator, making the file only a little bit bigger but keeping it easier to read?

Does this file _need_ to be human-readable, though? The individual BibTeX entries for the papers remain unchanged after all. (Genuine question; I personally prefer to copy over only the BibTeX entries I actually use, and take them from the website, so I don't use this file myself.)

**Comment by joanise** [comment]
> Does this file _need_ to be human-readable, though? The individual BibTeX entries for the papers remain unchanged after all. (Genuine question; I personally prefer to copy over only the BibTeX entries I actually use, and take them from the website, so I don't use this file myself.)

That's an excellent question, and one that made me hesitate to leave my comment. But I like to be able to use my own uploaded copy of this file in Overleaf to look at the information for papers I'm citing, so my preference would be to keep it human readable. I imagine the community will be divided on this, with a large number of people basically never looking at this file, and a number of other people consulting it as I do.

**Comment by mjpost** [comment]
Thanks for leaving your comment, we always welcome energy and feedback. There is still [the full bib with abstracts](https://aclanthology.org/anthology+abstracts.bib.gz), would this work for your purposes? If not, and if we do indeed view this change as a temporary fix, maybe I can restore spaces so long as we keep under 50 MB.

**Comment by joanise** [comment]
> Thanks for leaving your comment, we always welcome energy and feedback. There is still [the full bib with abstracts](https://aclanthology.org/anthology+abstracts.bib.gz), would this work for your purposes? If not, and if we do indeed view this change as a temporary fix, maybe I can restore spaces so long as we keep under 50 MB.

I know I'm being picky, but ideally the file I have in Overleaf would be reasonably human readable. I just checked and my suggestion would still keep the file under 44 MB at the moment so it would be OK.

Since it's temporary, though, I will concede that my suggestion is not important to implement now,

**Comment by mjpost** [comment]
It's fine. The change was easy and since I already knew it would work I don't mind. I just created a PR: #3053.

**Comment by mjpost** [comment]
Hi @joanise—I restored human readability. I also added a header with the date of generation. Let me know what you think.

**Comment by joanise** [comment]
@mjpost Nice, thanks, this is great! And excellent idea to add the date of generation.
